haskell-indexer-backend-core/haskell-indexer-backend-core.cabal

```diff
@@ -17,4 +17,5 @@ library
   exposed-modules:     Language.Haskell.Indexer.Backend.AnalysisOptions
   build-depends:       base >=4.8 && <5
                      , text
+  ghc-options:         -Wall
```

Comment:
Consider adding these, for completeness:

            "-Wall",
            "-Werror",
            "-Wcompat",
            "-Wincomplete-record-updates",
            "-Wincomplete-uni-patterns",
            "-Wredundant-constraints",

Reply:
Done to all cabal files.

Reply:
`-Werror` causes a build failure in the cabal v1 CI. Deleted (for now).

```
...
+++pwd
++ORIGDIR=/home/travis/build/google/haskell-indexer
++for dir in '$PACKAGES'
++cd /home/travis/build/google/haskell-indexer/haskell-indexer-pathutil/
++cabal check
Warning: The following errors will cause portability problems on other
environments:
Warning: 'ghc-options: -Wall -Werror' makes the package very easy to break
with future GHC versions because new GHC versions often add new warnings. Use
just 'ghc-options: -Wall' instead. Alternatively, if you want to use this,
make it conditional based on a Cabal configuration flag (with 'manual: True'
and 'default: False') and enable that flag during development.
Warning: Hackage would reject this package.
++'[' 2.4 == 1.16 ']'
```